novawallet/Common/QRCreation/Logo/QRCodeWithLogoFactory.swift

```diff
@@ -0,0 +1,260 @@
+            guard let self else { return }
+
+            dispatchInQueueWhenPossible(callbackQueue) {
+                do {
+                    let value = try partialQRWrapper.targetOperation.extractNoCancellableResultData()
+                    partialResultClosure(.success(value))
+                } catch {
+                    partialResultClosure(.failure(error))
+                }
+            }
+        }
+
+        let completeQRWrapper = createQRResultWrapper(
+            using: logoOperation,
+            payload: payload,
+            logoInfo: logoInfo,
+            qrSize: qrSize
+        )
+
+        completeQRWrapper.targetOperation.addDependency(partialQRWrapper.targetOperation)
+
+        return completeQRWrapper.insertingHead(operations: partialQRWrapper.allOperations)
+    }
+
+    func createCachedLogoQRWrapper(
+        cacheKey: String,
+        payload: Data,
+        logoInfo: IconInfo,
+        qrSize: CGSize
+    ) -> CompoundOperationWrapper<QRCreationResult> {
+        let logoOperation = iconRetrievingFactory.retrieveImageOperation(using: cacheKey)
+
+        return createQRResultWrapper(
+            using: logoOperation,
+            payload: payload,
+            logoInfo: logoInfo,
+            qrSize: qrSize
+        )
+    }
+
+    func createQRResultWrapper(
+        using logoOperation: BaseOperation<UIImage>?,
+        payload: Data,
+        logoInfo: IconInfo,
+        qrSize: CGSize
+    ) -> CompoundOperationWrapper<QRCreationResult> {
+        var qrImageWrapper: CompoundOperationWrapper<UIImage>
+        qrImageWrapper = OperationCombiningService.compoundNonOptionalWrapper(
+            operationManager: OperationManager(operationQueue: operationQueue)
+        ) { [weak self] in
+            guard let self else {
+                return .createWithError(BaseOperationError.parentOperationCancelled)
+            }
+
+            var updatedLogoInfo: IconInfo? = logoInfo
+
+            if let logoOperation {
+                do {
+                    let logoImage = try logoOperation.extractNoCancellableResultData()
+                    updatedLogoInfo = logoInfo.byChangingToLocal(logoImage)
+                } catch {
+                    logger.error(error.localizedDescription)
+                }
+            } else {
+                updatedLogoInfo = updatedLogoInfo?.withNoLogo()
+            }
+
+            let qrCodeOperation = QRWithLogoCreationOperation(
+                payload: payload,
+                qrSize: qrSize,
+                logoInfo: updatedLogoInfo
+            )
+
+            return CompoundOperationWrapper(targetOperation: qrCodeOperation)
+        }
+
+        if let logoOperation {
+            qrImageWrapper.addDependency(operations: [logoOperation])
+            qrImageWrapper = qrImageWrapper.insertingHead(operations: [logoOperation])
+        }
+
+        let resultMappingWrapper: CompoundOperationWrapper<QRCreationResult>
```

Comment:
It should be `ClosureOperation` since we just need to synchronously map the result